### PR TITLE
Added support for operation mode change and setting manual fan speed

### PR DESCRIPTION
**Pull request by JEZHDK**
Hi Jonas,

I have spend some time with your interface to danfoss air.
You are now able to change mode and set fan speed.

I would be cool to get the PR through, then i will work on the Home Assistant part afterwards.

Thank you.

### CODE REVIEW

**Comment by JonasPed** [comment]
Thanks. I will take a look at it as soon as I get a few hours of spare time.

**Comment by JonasPed** [comment]
Just tried to check out the code and run it. I get below error. Please ensure that it is working and I will take a look at it again. Current master works fine. Thanks. 

```
(venv) jonas-privat@jope-kit:~/development/pydanfossair/pydanfoss-air (JEZHDK/master)$ python -m pydanfossair --host 10.0.0.13
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/jonas-privat/development/pydanfossair/pydanfoss-air/pydanfossair/__main__.py", line 5, in <module>
    import danfossclient
ModuleNotFoundError: No module named 'danfossclient'
```

**Comment by JEZHDK** [comment]
What version of python are you using? And OS.

Older versions of python liked a . before the files to input.

**Comment by JonasPed** [comment]
Have tried with both Python 3.7 and 3.11 under Linux.

**Comment by JEZHDK** [comment]
Now i have tested on another PC to make sure. Works fine. Tested on Ubuntu and Win11.
Pulled from my PR, so 100% the same as you have. This is odd.

C:\Users\Jeppe\Desktop\test3>python pydanfossair --host 192.168.1.28
ReadCommand.exhaustTemperature: 9.46
ReadCommand.supplyTemperature: 18.87
ReadCommand.extractTemperature: 21.8
ReadCommand.outdoorTemperature: 7.02
ReadCommand.roomTemperature: 21.67
ReadCommand.roomTemperatureCalculated: 21.71
ReadCommand.humidity: 47.84313725490196
ReadCommand.bypass: False
ReadCommand.filterPercent: 33.72549019607843
ReadCommand.boost: True
ReadCommand.supply_fan_speed: 1738
ReadCommand.exhaust_fan_speed: 1819
ReadCommand.fan_step: 49
ReadCommand.away_mode: False
ReadCommand.battery_percent: 39.21568627450981
ReadCommand.automatic_bypass: False
ReadCommand.operation_mode: demand

**Comment by JonasPed** [comment]
It is because you are not executing it as a module. You are missing the `-m` argument.

**Comment by JEZHDK** [comment]
Works with -m now.

jezh@jezh-desktop:~/Desktop/pydanfoss-air$ python3 -m pydanfossair --host 192.168.1.28
ReadCommand.exhaustTemperature: 7.72
ReadCommand.supplyTemperature: 17.34
ReadCommand.extractTemperature: 20.4
ReadCommand.outdoorTemperature: 5.71
ReadCommand.roomTemperature: 22.29
ReadCommand.roomTemperatureCalculated: 21.61
ReadCommand.humidity: 36.86274509803921
ReadCommand.bypass: False
ReadCommand.filterPercent: 32.549019607843135
ReadCommand.boost: False
ReadCommand.supply_fan_speed: 1046
ReadCommand.exhaust_fan_speed: 1085
ReadCommand.fan_step: 20
ReadCommand.away_mode: False
ReadCommand.battery_percent: 38.8235294117647
ReadCommand.automatic_bypass: False
ReadCommand.operation_mode: demand

File: pydanfossair/danfossclient.py [hunk 1]
```diff
@@ -59,8 +62,30 @@ def _update_command(self, command, socket):
...
+                      UpdateCommand.set_fan_step_5,
+                      UpdateCommand.set_fan_step_6,
+                      UpdateCommand.set_fan_step_7,
+                      UpdateCommand.set_fan_step_8,
+                      UpdateCommand.set_fan_step_9,
+                      UpdateCommand.set_fan_step_10}:
+                      return self._read_byte(command, socket)
+
+
+        # TODO: Virker ikke som om man kan få denne exeption?
```
Comment (JonasPed) [inline pydanfossair/danfossclient.py]: It is there to catch programming errors where you only add a new command without doing all the necessary changes.

File: pydanfossair/danfossclient.py [hunk 2]
```diff
@@ -100,13 +127,11 @@ def _read_command(self, command, socket):
...
 
         if command == ReadCommand.operation_mode:
-            return_value = OperationMode(self._read_byte(command, socket))
+            return_value = OperationMode(self._read_byte(command, socket)).name
 
         if return_value is not None:
             return return_value
 
-        raise Exception("Unknown command: {0}".format(command))
-
```
Comment (JonasPed) [inline pydanfossair/danfossclient.py]: I don't think you should remove this line. It is there to catch programming errors where you only add a new command without doing all the necessary changes.

File: pydanfossair/danfossclient.py [hunk 3]
```diff
@@ -38,8 +38,8 @@ def _command(self, command, socket):
 
         if isinstance(command, UpdateCommand):
             return self._update_command(command, socket)
-
-        raise Exception("Not yet implemented")
+        else:
+            raise Exception("Not yet implemented")
```
Comment (JonasPed) [inline pydanfossair/danfossclient.py:42]: You should wither keep the raise Exception statement outside else or use 
```
if
elif
else
```

construction.

File: README.md [hunk 4]
```diff
@@ -15,3 +15,15 @@ Python module and client for Danfoss Air HRV systems.
 - bypass_off
 - automatic_bypass_on
 - automatic_bypass_off
+- operation_mode_demand
+- operation_mode_program
+- operation_mode_manual
+- set_fan_step_2
+- set_fan_step_3
```
Comment (JonasPed) [inline README.md:28]: What about set_fan_step_1?

**Review by JonasPed — CHANGES_REQUESTED** [review]
Overall it looks good. Just a couple of minor comments.

File: README.md (hunk 4 above)
Comment (JEZHDK) [reply README.md:28]: Added.

File: pydanfossair/danfossclient.py (hunk 1 above)
Comment (JEZHDK) [reply pydanfossair/danfossclient.py]: Super, have removed the comment.

File: pydanfossair/danfossclient.py (hunk 2 above)
Comment (JEZHDK) [reply pydanfossair/danfossclient.py]: Added it again.

File: pydanfossair/danfossclient.py (hunk 3 above)
Comment (JEZHDK) [reply pydanfossair/danfossclient.py:42]: Yes correct, fixed.

**Comment by JEZHDK** [comment]
Hi Jonas,
Thank you for your time.
I have fixed the comments (i hope), and i hope i have done everything right in GitHub.

**Comment by JonasPed** [comment]
Version 0.2.0 just published.

**Pull request merged**